### PR TITLE
tag: allow entering a reason

### PR DESCRIPTION
**Pull request by siddharthvp**
Allow entering a reason to be appended in the edit summary. 

Unlike adding of tags, it is typically expected that a reason be given while removing tags. Truth be told, the untag feature (#485) remains incomplete without this. I don't think I have seen the untag feature being used any more than 1-2 times.

Demo of usecase: [Special:Diff/945510112](https://en.wikipedia.org/w/index.php?title=Via.com&diff=945510112)

Currently the reason field is shown all the time. I'm open to feedback on this. Would it be better to show the reason field only when one or more tags have been de-selected?

### CODE REVIEW

**Comment by Amorymeltzer** [comment]
I think it's fine to have all the time.  It's at the bottom so out of the way, and I think it'd be nice if folks used it when adding as well.

File: modules/friendlytag.js [hunk 1]
```diff
@@ -129,8 +129,15 @@ Twinkle.tag.callback = function friendlytagCallback() {
...
 				]
-			}
-			);
+			});
+
+			form.append({
+				type: 'input',
+				label: 'Reason',
+				name: 'reason',
+				tooltip: 'Optional reason to be appended in edit summary. Usually needed only while removing tags',
```
Comment (Amorymeltzer) [inline modules/friendlytag.js]:
```suggestion
				tooltip: 'Optional reason to be appended in edit summary. Recommended when removing tags.',
```

File: modules/friendlytag.js [hunk 2]
```diff
@@ -2014,6 +2023,7 @@ Twinkle.tag.callback.evaluate = function friendlytagCallbackEvaluate(e) {
 		case 'article':
 			params.tagsToRemove = form.getUnchecked('alreadyPresentArticleTags') || [];
 			params.tagsToRemain = form.getChecked('alreadyPresentArticleTags') || [];
+			params.reason = form.reason.value;
```
[the remark this answers is not in the document]
Comment (siddharthvp) [reply modules/friendlytag.js]: meh, I guess it's time for `Morebits.quickForm.getInputData` to finally see the light of the day, which should get rid of needing to do all this by hand!

**Comment by siddharthvp** [comment]
> I also think we should change pntPage's use of params.reason to params.pntreason or something

handled in a 2nd commit, a bit differently.

**Pull request merged**